rest/route/github.go

```diff
@@ -263,11 +264,107 @@ func (gh *githubHookApi) Run(ctx context.Context) gimlet.Responder {
 		if event.GetAction() == githubActionChecksRequested {
 			return gh.handleMergeGroupChecksRequested(event)
 		}
+
+	case *github.CheckRunEvent:
+		if event.GetAction() == githubActionRerequested {
+			return gh.handleCheckRunRerequested(ctx, event)
+		}
 	}
 
 	return gimlet.NewJSONResponse(struct{}{})
 }
 
+func (gh *githubHookApi) handleCheckRunRerequested(ctx context.Context, event *github.CheckRunEvent) gimlet.Responder {
```

Comment:
Can we add a comment either here or higher up explaining that this will handle a user clicking the restart button on the check run in GitHub?

Reply:
added